<launch>
  <arg name="gui" default="true" />
  <arg name="sim" default="false" />


  <param name="/use_sim_time" value="$(arg sim)" />
  
  <!-- contains nd_supervisor(MsgStatus), nd_velocity, nd_geometry.py(image processing), nd_moments.py, usb_cam_node,image_view -->
  <include file="$(find camera_measures)/launch/camera_measures.launch">
    <arg name="gui" value="$(arg gui)" />
  </include>

  <!-- laser control parameters-->
  <rosparam command="load" file="$(find laam_laser_control)/config/control_parameters.yaml" />

  <!-- received processed image and published controlled power value-->
  <node name="control" pkg="laam_laser_control" type="nd_control.py">
    <param name="power_min" type="double" value="0" />
    <param name="power_max" type="double" value="1500" />
  </node>

  <node name="control_panel" pkg="laam_laser_control" type="qt_control.py" if="$(arg gui)" />

  <node name="labjack_power_output" pkg="labjack_ros" type="nd_labjack.py" respawn="true"/>

  <!-- pid parameter adaptive calculation -->
  <node name="pid_vrft_calculation" pkg="laam_laser_control" type="nd_adaptive_pid_vrft.py" />

</launch>
